app/controllers/RestaurantController.java

```diff
@@ -75,11 +80,15 @@ public Result deleteRestaurant(String id) {
 	@Transactional(readOnly = true)
 	public Result getAllRestaurants() {
 		String cityFilter = request().getQueryString(CITY_FILTER);
+		String cuisine = request().getQueryString(CUISINE_FILTER);
 		return wrapForPublic(() -> this.service.findRestaurantsWithFilter(
 				RestaurantFilter.createFilter()
 						.setPageNumber(getQueryInt(request().getQueryString(PAGE_NUMBER), DEFAULT_PAGE_NUMBER))
 						.setPageSize(getQueryInt(request().getQueryString(PAGE_SIZE), DEFAULT_PAGE_SIZE))
 						.setNameFilter(request().getQueryString(NAME_FILTER))
+						.setCuisineFilter(new ArrayList<String>(Arrays.asList(cuisine)))
```

Comment:
How exactly is this supposed to work? `cuisine` is a `String` as defines on line 83. `Arrays.asList` takes an Array as the only input parameter. It also returns a `List<T>` . Why then repackage that into a `new ArrayList()` ?